File: Solutions/974. Subarray Sums Divisible by K/974.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        ans = 0
        hm = {0: 1}
        sum = 0
        for i in nums:
            sum += i
            rem = sum % k
            if rem in hm:
                ans += hm[rem]
            hm[rem] = hm.get(rem, 0) + 1
        return ans
        

File: Solutions/974. Subarray Sums Divisible by K/974.java
class Solution {
    public int subarraysDivByK(int[] nums, int k) {
        int ans = 0;
        int sum = 0;
        HashMap<Integer, Integer> hm = new HashMap<>();
        hm.put(0, 1);
        int rem = 0;
        for(int i : nums){
            sum += i;
            rem = (sum + k) % k;
            if(hm.containsKey(rem)){
                ans += hm.get(rem);
            }
            hm.put(rem, hm.getOrDefault(rem, 0) + 1);
        }
        return ans;
    }
}